keyboards/melgeek/z70ultra/rev1/rev1.c

```diff
@@ -0,0 +1,169 @@
+		{0, 32},  {16, 32},  {32, 32}, {48, 32}, {64, 32}, {80, 32}, {96, 32}, {112, 32}, {128, 32}, {144, 32}, {160, 32}, {176, 32},            {192, 32}, 
+		{0, 48},             {16, 48}, {32, 48}, {48, 48}, {64, 48}, {80, 48}, {96, 48},  {112, 48}, {128, 48}, {144, 48}, {160, 48}, {176, 48},
+		{0, 64},  {16, 64},  {32, 64},           {48, 64}, {64, 64}, {80, 64},                        {96, 64}, {112, 64}, 
+		                                                             {0, 80},  {16, 80},  {32, 80},   {48, 80}, {64, 80}, {80, 80},   {96, 80}, {112, 80}, 
+	}, {
+	       4,       4,         4,        4,        4,        4,        4,        4,          4,         4,         4,          4,        4,         4,
+		   4,       4,         4,        4,        4,        4,        4,        4,          4,         4,         4,          4,        4,         4,
+		   4,       4,         4,        4,        4,        4,        4,        4,          4,         4,         4,          4,                   4,
+		   4,                  4,        4,        4,        4,        4,        4,          4,         4,         4,          4,        4,
+		   4,       4,         4,                  4,        4,        4,                               4,         4,
+		                                                               4,        4,          4,         4,         4,          4,        4,         4,
+	}
+};
+
+
+bool led_update_kb(led_t led_state) {
+    if (led_update_user(led_state)) {
+        if (led_state.caps_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[71], 0xff, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[72], 0xff, 0x00, 0x00);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[71], 0x00, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[72], 0x00, 0x00, 0x00);
+        }
+
+        if (led_state.num_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[70], 0x00, 0xff, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[73], 0x00, 0xff, 0x00);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[70], 0x00, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[73], 0x00, 0x00, 0x00);
+        }
+
+        if (led_state.scroll_lock) {
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[69], 0x00, 0x00, 0xff);
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[74], 0x00, 0x00, 0xff);
+        } else {
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[69], 0x00, 0x00, 0x00);
+            IS31FL3741_set_pwm_buffer(&g_is31_leds[74], 0x00, 0x00, 0x00);
+        }
+    }
+    return true;
+}
+
+void matrix_init_kb(void) {
+    for (int i = AXIS_LED_TOTAL; i < DRIVER_LED_TOTAL; ++i) {
+        is31_led led = g_is31_leds[i];
+        IS31FL3741_set_scaling_registers(&led, 0xFF, 0xFF, 0xFF);
+    }
+
+    IS31FL3741_update_led_control_registers(DRIVER_ADDR_1, 0);
+
+    matrix_init_user();
+}
```

Comment:
```suggestion
```

These should be included in the matrix itself. The  "indicators" flag exists for this reason.